src/libsyntax/parse/parser.rs

```diff
@@ -258,6 +258,14 @@ pub struct Parser<'a> {
     crate unclosed_delims: Vec<UnmatchedBrace>,
 }
 
+impl<'a> Drop for Parser<'a> {
+    fn drop(&mut self) {
+        if !self.unclosed_delims.is_empty() {
+            let diag = self.diagnostic();
+            emit_unclosed_delims(&mut self.unclosed_delims, diag);
```

Comment:
With logic like this, seems like all the remaining calls to `emit_unclosed_delims` can be removed.

Reply:
Sadly the ones that get called after `parse_stream_from_source_str` can't be removed (they are using a `ParseSess` instead of a `Parser`), but given that every call of it does it it can be added into its body to have it called in one place for those.